File: Technical Tranining/day3.py
# Define values
rows = 4
cols = 4

# Dec and init a 2D array
arr = [[0] * cols for i in range(rows)]

for i in range(rows):
    for j in range(cols):
        arr[i][j] = i

# Print the matrix
print("Original Matrix:")
for i in range(rows):
    for j in range(cols):
        print(arr[i][j], end=" ")
    print()

# Access and modify an element
print("\nElement at [2][2]:", arr[2][2])

arr[2][2] = 10

print("\nModified Matrix:")
for i in range(rows):
    for j in range(cols):
        print(arr[i][j], end=" ")
    print()

# Sum the elements of each row
print("\nRow Sums:")
for i in range(rows):
    total = 0
    for j in range(cols):
        total += arr[i][j]
    print("Sum of row", i, ":", total)

# Transpose of the matrix
transpose = [[0] * rows for i in range(cols)]

for i in range(rows):
    for j in range(cols):
        transpose[j][i] = arr[i][j]

print("\nTranspose Matrix:")
for i in range(cols):
    for j in range(rows):
        print(transpose[i][j], end=" ")
    print()

File: Technical Tranining/day3.java
public class day3 {
    public static void main(String[] args) {
        
        //Define the dimensions
        int row=4,col=4;

        //Declare and Initialize a 2d ARRAY
        int[][] arr = new int[row][col];
        for(int i =0;i<row;i++)
        {
            for(int j=0;j<col;j++)
            {
                arr[i][j] = i;
            }
        }

        //Print the matrix
        for(int i=0;i<row;i++)
        {
            for(int j=0;j<col;j++)
            {
                System.out.print(arr[i][j] + " ");
            }
            System.out.println();
        }

        //Sum the elements of each row
        
        for(int i =0;i<row;i++)
        {
            int sum =0;
            for(int j=0;j<col;j++)
            {
                sum+=arr[i][j];
            }
            System.out.println("Sum of row"+ " "+ i +" : "+ sum);
        }

        //Perform transpose of matrix
        int[][] transpose = new int[arr.length][arr[0].length];

        for(int i =0;i<row;i++)
        {
            for(int j=0;j<col;j++)
            {
                transpose[j][i] = arr[i][j];
            }
        }

        for(int i=0;i<row;i++)
        {
            for(int j=0;j<col;j++)
            {
                System.out.print(transpose[i][j] + " ");
            }
            System.out.println();
        }


    }
    
}
